eng/copilot/testTranslation.js

```diff
@@ -0,0 +1,95 @@
+        if (language !== changedFileLanguage) {
+            const languageFiles = changedFiles.filter(file => file.filename.includes(".cs") || file.filename.includes(".py") || file.filename.includes(".js") || file.filename.includes(".go") || file.filename.includes(".java"));
+            const translationPromises = languageFiles.map(async (file) => {
+                console.log(`[${changedFileLanguage} => ${language}]\tstart translating ${file.filename} ...`);
+                const dpResponse = await translate(file, accessSession.session_id, accessSession.access_token, language);
+                console.log(`[${changedFileLanguage} => ${language}]\t${file.filename} translation complete`);
+                return dpResponse;
+            });
+            const translatedFilesForLanguage = await Promise.all(translationPromises);
+            translatedFiles.push(...translatedFilesForLanguage);
+        }
+    });
+    await Promise.all(translationPromises);
+
+
+    //prepare for github commit
+    const sha = await getLatestCommitSha(targetRepoOwner, targetRepo);
+    let changeSha = await createChangeBranch(targetRepoOwner, targetRepo, sha);
+
+    //stash files -> commit -> push
+    const blobs = await createBlob(targetRepoOwner, targetRepo, translatedFiles);
+    const treeSha = await createTree(targetRepoOwner, targetRepo, blobs, changeSha);
+    const commitSha = await createCommit(targetRepoOwner, targetRepo, treeSha, changeSha);
+    await updateBranch(targetRepoOwner, targetRepo, commitSha);
+
+    //create pr
+    if (changeSha === sha) {
+        console.log("Branch already exists. Skipping PR creation.");
+    } else {
+        await createPR(targetRepoOwner, targetRepo);
+    }
+}
+
+async function translate(file, sessionId, accessToken, targetLanguage) {
+    const query = `
+                Below is a file change patch from github pull request.
+                Go through this file name and file patch then translate the file content to ${targetLanguage}.
+                Return the response in stringfied json format, with fileName and fileContent.
+                The file name should begin with path tests/integration-tests/${targetLanguage}
+                For your response, use same environment variable(hub, connection string, messages etc.) as the original file.
+                Do not omit any file content. The response should contain as many tests as the original document.
+
+                Include these import statement as necessary in your response:
+                ###
+                ${prompt[targetLanguage]}
+                ###
+                File Name: ###${file.filename}###
+                File patch:###${file.patch}###`;
+    try {
+        while (true) {
+            const dpResponse = await fetchDeepPromptWithQuery(query, sessionId, accessToken);
+            if (dpResponse && dpResponse.includes("fileName") && dpResponse.includes("fileContent")) {
```

Comment:
`includes` sounds dangerous, how about adding some hint into the prompt, to let it start with some special words to indicate a `success`?

Reply:
since this instruction is included in the query, as long as deep prompt group does not modify their output, the result should always contains "fileName" and "fileContent", otherwise, we may consider it might be "I can only assist with development topic" or "deep prompt busy" message